pkg/operator/ceph/object/notification/controller.go

```diff
@@ -237,26 +197,46 @@ func (r *ReconcileNotifications) updateCephBucketNotification(notification *ceph
 			return reconcile.Result{}, err
 		}
 
-		// provision the notification
-		provisioner := Provisioner{
-			Client:           r.client,
-			Context:          r.context,
-			ClusterInfo:      r.clusterInfo,
-			ClusterSpec:      r.clusterSpec,
-			opManagerContext: r.opManagerContext,
-		}
-		session, err := provisioner.createSession(
-			ob.Spec.AdditionalState["cephUser"],
-			objectStoreName,
-		)
+		s3Agent, err := NewS3Agent(r.opManagerContext, r.context, clusterInfo, &cephCluster.Spec, ob.Spec.AdditionalState["cephUser"], objectStoreName)
 		if err != nil {
-			return reconcile.Result{}, errors.Wrapf(err, "failed to create session for CephBucketNotification provisioning %q for ObjectBucketClaims %q", bnName, bucketName)
+			return reconcile.Result{}, errors.Wrapf(err, "failed to create S3 agent for CephBucketNotification provisioning %q for ObjectBucketClaims %q", bnName, bucketName)
 		}
-		err = provisioner.Create(&ob, topicARN, notification, session)
+		err = Create(&ob, topicARN, notification, s3Agent)
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "failed to provision CephBucketNotification %q for ObjectBucketClaims %q", bnName, bucketName)
 		}
 	}
 
 	return reconcile.Result{}, nil
 }
+
+func getCephObjectStoreName(ob bktv1alpha1.ObjectBucket) (objectStoreName types.NamespacedName, err error) {
+	// prase the following string: <prefix>-rgw-<store>.<namespace>.svc
+	// to ge the object store name and namespace
+	logger.Debugf("BucketHost of %q is %q",
+		types.NamespacedName{Name: ob.Name, Namespace: ob.Namespace}.String(),
+		ob.Spec.Endpoint.BucketHost,
+	)
+	parsedBucketHost := strings.Split(ob.Spec.Endpoint.BucketHost, ".")
+	if len(parsedBucketHost) < 3 {
+		err = errors.Errorf("malformed BucketHost %q", ob.Spec.Endpoint.BucketHost)
+		return
+	}
+	parsedSubdomain := strings.Split(parsedBucketHost[0], "-rgw-")
+	if len(parsedSubdomain) != 2 {
+		err = errors.Errorf("malformed BucketHost subdomain %q", parsedBucketHost[0])
+		return
+	}
+	objectStoreName.Namespace = parsedBucketHost[1]
+	objectStoreName.Name = parsedSubdomain[1]
```

Comment:
this was my initial implementation, but at some point, the tests started to fail because the integration framework was using a different prefix.
in the latest runs, i don't see that anymore. also seems like this is hardcoded in the `BuildDomainName()` in rgw.go.
so, i will revert the code to assume the hardcoded prefix (and also add a test so we make sure this is won't break i in the future).